common/common.go

```diff
@@ -154,10 +155,32 @@ func (l *PrivateNetList) String() string {
 
 func (l *PrivateNetList) Set(value string) error {
 	if l.mapping == nil {
-		l.mapping = make(map[string]bool)
+		l.mapping = make(map[string]string)
 	}
 	for _, s := range strings.Split(value, ",") {
-		l.mapping[s] = true
+		netArgsPair := strings.Split(s, ":")
+		netName := netArgsPair[0]
+
+		if netName == "" {
+			return fmt.Errorf("netname must not be empty")
+		}
+		if _, duplicate := l.mapping[netName]; duplicate {
+			return fmt.Errorf("found duplicate netname %q", netName)
+		}
+
+		switch {
+		case len(netArgsPair) == 1:
+			l.mapping[netName] = ""
+		case len(netArgsPair) == 2:
+			if netName == "all" || strings.HasPrefix(netName, "default") {
+				return fmt.Errorf("arguments are not supported by special netname %q", netName)
```

Comment:
I see. What about "default-restricted"? We rely on that to be able to communicate with the metadata service. If we allow to override it we give the user the possibility to break that, but on the other hand he could also not run the metadata service at all :-)